### PR TITLE
6.0 deprecations - various

### PR DESCRIPTION
**Pull request by gasman**
Handle most of the remaining deprecations for Wagtail 6.0. Incorporates ~#11123~, ~#11138~, ~#11146~, ~#11154~

A few front-end items have been left to be picked up by someone else (@lb- / @thibaudcolas?), as I wasn't familiar enough with the code to be confident of removing the right bits and no more or less...

- [x] `client/src/controllers/SwapController.ts` / `client/src/controllers/SwapController.test.js` (removing fallback on global `headerSearch` variables)
- [x] `client/src/controllers/TagController.ts` (removing global `initTagField` function)
- [x] `client/src/controllers/ProgressController.ts` (removing legacy `cancelSpinner` function)
- [ ] Removal of icon font and deprecated / renamed icons - `client/scss/settings/_variables.icons.scss`, `client/scss/components/_icons.scss`, `wagtail/admin/tests/test_templatetags.py`, `wagtail/admin/templatetags/wagtailadmin_tags.py`, `wagtail/admin/static_src/wagtailadmin/fonts/wagtail.woff` and probably various other bits of CSS and tooling

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)

### CODE REVIEW

**Comment by lb-** [comment]
Related PR up for JS/Stimulus removals - https://github.com/wagtail/wagtail/pull/11161

I have not looked at the icons though.

File: wagtail/contrib/modeladmin/__init__.py
Comment (laymonage) [inline wagtail/contrib/modeladmin/__init__.py:1]: Do we need to remove the Transifex config as well?

https://github.com/wagtail/wagtail/blob/aae3ce1c84402c42ff09b92ec2f9e9aac5c59797/.tx/config#L53-L57

File: docs/topics/streamfield.md [hunk 1]
```diff
@@ -334,7 +334,7 @@ body = StreamField([
     ('image', ImageChooserBlock()),
 ], block_counts={
     'heading': {'min_num': 1, 'max_num': 3},
-}, use_json_field=True)
+})
```
Comment (laymonage) [inline docs/topics/streamfield.md:337]: There's another one further down below after #11135 was merged

**Review by laymonage — APPROVED** [review]
Minor comments but otherwise this looks good to me!

**Pull request merged**